### ISSUE TITLE
Datasource Contents provided by plugins are not accepted as valid choice by REST API

### ISSUE THREAD

**Issue by progala** [nautobot/nautobot#1581]
### Environment
* Python version: Python 3.7.12
* Nautobot version: v1.2.0

### Steps to Reproduce
1. Install Golden Config Plugin

2. Send POST request to `{{NAUTOBOT_URL}}/api/extras/git-repositories/` with the following payload:
```
{
        "name": "gctest",
        "slug": "gctest",
        "remote_url": "https://gh.example/demo-gc-test",
        "provided_contents": [
            "nautobot_golden_config.backupconfig"
        ]
}
```

### Expected Behavior

POST request succeeds and a new Git Repository object is created.

### Observed Behavior

API returns error stating that the `nautobot_golden_config.backupconfig` is not a valid choice for `provided_contents`.

```
{
    "provided_contents": [
        "\"nautobot_golden_config.backupconfig\" is not a valid choice."
    ]
}
```

Poking around in the local instance of Nautobot w/ GC plugin, via nbshell I can see that the datasource contents coming from Golden Config show up in the registry. REST serializer however doesn't see those.

It appears the choices used by serializer are evaluated once, before plugins are loaded.

```
>>> from nautobot.extras.registry import registry
>>> pprint([gc.content_identifier for gc in registry["datasource_contents"]["extras.gitrepository"]])
['extras.configcontextschema',
 'extras.configcontext',
 'extras.job',
 'extras.exporttemplate',
 'nautobot_golden_config.intendedconfigs',
 'nautobot_golden_config.jinjatemplate',
 'nautobot_golden_config.backupconfigs',
 'nautobot_golden_config.pluginproperties']

>>> from nautobot.extras.api.serializers import GitRepositorySerializer
>>> GitRepositorySerializer._declared_fields["provided_contents"]
MultipleChoiceJSONField(allow_blank=True, choices=
[('extras.configcontext', 'config contexts'),
 ('extras.configcontextschema', 'config context schemas'),
 ('extras.exporttemplate', 'export templates'),
 ('extras.job', 'jobs')],
 required=False)
```

**Comment by glennmatthews** [comment]
I think the listing is currently evaluated at serializer class declaration time. It might be possible to change that into a deferred callback instead

**Comment by jathanism** [comment]
Evidence of fix in #1588. It's now pointing to a `lambda` and accessing `.choices` will display that the choices do in fact update dynamically.

```python
In [13]: GitRepositorySerializer._declared_fields["provided_contents"]
Out[13]: MultipleChoiceJSONField(allow_blank=True, choices=<function GitRepositorySerializer.<lambda>>, required=False)

In [14]: GitRepositorySerializer._declared_fields["provided_contents"].choices
Out[14]:
OrderedDict([('extras.configcontext', 'config contexts'),
             ('extras.configcontextschema', 'config context schemas'),
             ('extras.exporttemplate', 'export templates'),
             ('extras.job', 'jobs'),
             ('nautobot_golden_config.backupconfigs', 'backup configs'),
             ('nautobot_golden_config.intendedconfigs', 'intended configs'),
             ('nautobot_golden_config.jinjatemplate', 'jinja templates')])

```
